File: FRONT-END/lib/dto/chat/request/chat_dto.dart
class ChatDTO {
  final String? conversationId;
  final String prompt;
  final String? userType;

  ChatDTO({
    this.conversationId,
    required this.prompt,
    this.userType,
  });

  bool needsConversationId() {
    return conversationId == null || conversationId!.trim().isEmpty;
  }

  Map<String, dynamic> toJson() {
    final json = {
      if (conversationId != null) 'conversationId': conversationId,
      'prompt': prompt,
      if (userType != null) 'userType': userType,
    };
    print('📤 ChatDTO: Enviando al backend: $json');
    return json;
  }
}

File: BACK-END/KuentecoChat/src/main/java/com/example/back_end/dto/request/ChatDTO.java
package com.example.back_end.dto.request;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ChatDTO {

    private String conversationId;

    @NotBlank(message = "The prompt must be defined")
    private String prompt;

    public boolean needsConversationId() {
        return conversationId == null || conversationId.trim().isEmpty();
    }
}
